west/migrate.py

```diff
@@ -3,53 +3,60 @@
 import logging
 import os.path
 
+from west import core
 from west import db
 from west import exceptions
 from west import files
 from west import runner
 from west import utils
-from west import west
 from west.settings import settings
 
 logger = logging.getLogger(__name__)
 
 
 def migrate():
-    # build miration plan
-    migration_plan = west.build_migration_plan()
 
-    if migration_plan is None:
+    logger.info("Starting migrations")
+    if not db.is_schema_initialized():
+        logger.info("Migration table is empty, loading a schema")
         # schema not inited
-        schema_version = west.get_schema_version()
+        schema_version = core.get_best_schema_version()
         init_schema(schema_version)
-        # reload migration_plan
-        migration_plan = west.build_migration_plan(schema_version)
 
     # play migrations
     print("Apply migrations")
-    for plan in migration_plan["plans"]:
+    width = 40
+    for plan in core.build_migration_plan():
         version = plan["version"]
-        logger.info(version)
+        logger.info("Processing version %s", version)
         for mig, applied, path, is_manual in plan["plan"]:
+            logger.debug(
+                "Processing migration %(mig)s, applied: %(applied)s, "
+                "path: %(path)s, manual: %(is_manual)s",
+                {"mig": mig, "applied": applied, "path": path, "is_manual": is_manual},
+            )
             title = mig
             if is_manual:
                 title += " (manual)"
+            title += " "
             if applied:
-                print("  {} already applied".format(title))
+                print("  {:.<{}} Already applied".format(title, width))
             else:
-                print("  Applying {}...".format(title))
+                print("  {:.<{}} Applying ...".format(title, width))
```

Comment:
the output is : 
 ```
Apply migrations
  1.0-0-version-dml.sql .................. Applying ...
  1.0-author-1-ddl.sql ................... Applying ...
  1.0-author-2-dml.sql ................... Applying ...
  1.0-book-1-ddl.sql ..................... Applying ...
  1.0-book-2-dml.sql ..................... Applying ...
  1.1-0-version-dml.sql .................. Applying ...
  1.1-add-num-pages-1-ddl.sql ............ Applying ...
  1.1-add-num-pages-2-dml.sql (manual) ... Applying ...
  1.1-index-ddl.sql ...................... Applying ...
  1.2-0-version-dml.sql .................. Applying ...
  1.2-remove-author-dob-ddl.sql .......... Applying ...
  1.2-rename-num-pages-ddl.sql ........... Applying ...
```

Reply:
Yep. I find it nice. Don't you ?

Reply:
no :)
I think I'm expecting something to finish the `...` at the end (like a :ballot_box_with_check:)
I don't know if we could print `☐` then `\b☑` :D

Reply:
or maybe we could have an output that looks more like the `show-migrations` output

Reply:
> I don't know if we could print ☐ then \b☑ :D

You want _kikoolol_, I'll get you some _kikoolol_ :)

https://github.com/peopledoc/west/pull/7/commits/4aaee98b5233ddd158777fe09237e35a290bdb1a brings a style module that can be completely disengaged, and offers checkbox context managers, where the chexbox is first drawn unchecked and in red, and then redrawn checked and in green :)

![screenshot from 2018-06-18 23-23-35](https://user-images.githubusercontent.com/1457576/41563130-fee91f38-734e-11e8-85f1-d33ab6f88d2e.png)
![screenshot from 2018-06-18 23-22-58](https://user-images.githubusercontent.com/1457576/41563131-ff03aace-734e-11e8-837c-3b2b9d1b7fda.png)